RetrospectiveExtension.Frontend/dal/reflectBackendService.tsx

```diff
@@ -60,31 +61,34 @@ class ReflectBackendService {
       return false;
     }
 
-    this._connectionAvailable = await this._signalRConnection.start()
-    .then(() => true)
-    .catch((error) => {
+    try {
+      await this._signalRConnection.start();
+      this._connectionAvailable = true;
+    }
+    catch (error) {
       console.error('Error when trying to start signalR connection: ', error);
       console.debug('Unable to establish signalR connection, live syncing will be affected.');
-      return false;
-    });
+      this._connectionAvailable = false;
+    }
 
     return this._connectionAvailable;
   }
 
   private retrieveValidToken = (that = this) => {
     if (that._tokenExpiry && moment().isBefore(that._tokenExpiry)) {
-      return that._appToken.token;
+      return that._appToken;
     }
 
-    return Promise.resolve(VSS.getAppToken().then((appToken) => {
+    return Promise.resolve(getAppToken().then((appToken) => {
       that._appToken = appToken;
 
-      const tokenData = jsonwebtoken.decode(that._appToken.token, {json: true});
-      if (typeof tokenData === 'object') {
+      const tokenData = decodeJwt(that._appToken);
+      if (tokenData) {
         that._tokenExpiry = moment.unix(tokenData.exp).toDate();
-        return that._appToken.token;
+        return that._appToken;
       }
 
+      // TODO (phongcao) : appInsightsClient.trackException(new Error(e.message));
```

Comment:
@phongcao  - just as a headsup this TODO got in

Reply:
yeah it's there as a reminder for us to add AppInsights to this file in the future.

Reply:
Resolved as will not fixed for now.